lib/Dashboard/CalendarWidget.php

```diff
@@ -143,31 +154,65 @@ public function load(): void {
+		// This is hw JS does it:
+		//          const start = dateFactory()
+		//			const end = dateFactory()
+		//			end.setDate(end.getDate() + 14)
+		//          const startOfToday = moment(start).startOf('day').toDate()
+		// get all vevents in this time range
+		// if "show tasks" is enabled, get all todos in the time range
+		// sort events by time
+		// filter events by:
+		// .filter(event => !event.classNames.includes('fc-event-nc-task-completed'))
+		// .filter(event => !event.classNames.includes('fc-event-nc-cancelled'))
+		// filter out all events that are before the start of the day:
+		// .filter(event => filterBefore.getTime() <= event.start.getTime())
+		// decorate the items with url / task url, colour, etc
+
 		$calendars = $this->calendarManager->getCalendarsForPrincipal('principals/users/' . $userId);
 		$count = count($calendars);
 		if ($count === 0) {
 			return [];
 		}
-		$dateTime = (new DateTimeImmutable())->setTimestamp($this->timeFactory->getTime());
-		$inTwoWeeks = $dateTime->add(new DateInterval('P14D'));
-		$options = [
-			'timerange' => [
-				'start' => $dateTime,
-				'end' => $inTwoWeeks,
-			]
-		];
+
 		$widgetItems = [];
 		foreach ($calendars as $calendar) {
-			$searchResult = $calendar->search('', [], $options, $limit);
-			foreach ($searchResult as $calendarEvent) {
-				/** @var DateTimeImmutable $startDate */
-				$startDate = $calendarEvent['objects'][0]['DTSTART'][0];
+			$timezone = null;
+			if($calendar instanceof CalendarImpl) {
+				/** @var VTimeZone $vTimezone */
+				$timezone = $calendar->getCalendarTimezoneString();
+			}
+			// make sure to include all day events
+			$startTimeWithTimezone =  (!empty($timezone)) ? $this->timeFactory->getDateTime('today', new \DateTimeZone($timezone) ?? null) : $this->timeFactory->getDateTime('today');
+			$endDate = clone $startTimeWithTimezone;
+			$endDate->modify('+15 days');
+			$options = [
+				'timerange' => [
+					'start' => $startTimeWithTimezone,
+					'end' => $endDate,
+				],
+				'sort_asc' => [
+					'firstoccurence'
+				]
+			];
+			$searchResults = $calendar->search('', [], $options, $limit);
+			foreach ($searchResults as $calendarEvent) {
+				$dtStart = DateTime::createFromImmutable($calendarEvent['objects'][0]['DTSTART'][0]);
+				$dtEnd = (isset($calendarEvent['objects'][0]['DTEND'])) ? DateTime::createFromImmutable($calendarEvent['objects'][0]['DTEND'][0]) : null;
+				if(!empty($dtEnd) && $dtStart->diff($dtEnd)->days >= 1) {
+					// All day (and longer)
+					$timeString = $this->dateTimeFormatter->formatDateSpan($dtStart);
+				} elseif(!empty($dtEnd)) {
+					$timeString = $this->dateTimeFormatter->formatDateTime($dtStart, 'short', 'short') . ' - ' . $this->dateTimeFormatter->formatTime($dtEnd, 'short');
+				} else {
+					$timeString = $this->dateTimeFormatter->formatTimeSpan($dtStart, $startTimeWithTimezone);
+				}
 				$widget = new WidgetItem(
 					$calendarEvent['objects'][0]['SUMMARY'][0] ?? 'New Event',
```

Comment:
**thought (non-blocking):** Unrelated, but the fallback should rather be named `Untitled event`.

Reply:
right, then it's in line with the frontend. nice catch